nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/comment/AnnotationAdapter.java

```diff
@@ -108,4 +114,17 @@ public Response deleteExternalAnnotation(@PathParam("entityId") String entityId)
         return Response.noContent().build();
     }
 
+    protected List<Annotation> getAllAnnotations(AnnotationService annotationService, String documentId, String xpath) {
+        CoreSession session = getContext().getCoreSession();
+        List<Annotation> allAnnotations = new ArrayList<>();
+        List<Annotation> childrenAnnotations = annotationService.getAnnotations(session, documentId, xpath);
+        for (Annotation child : childrenAnnotations) {
+            if (annotationService.getAnnotations(session, child.getId(), xpath).size() > 0) {
+                allAnnotations.addAll(getAllAnnotations(annotationService, child.getId(), xpath));
```

Comment:
Children are not necessarily `Annotation`, they will commonly be `Comment`. This means to put this logic in `CommentManager` but this would mean to instantiate `Annotation` from there, not sure we want this right now.
For the our need, maybe something like this on search endpoint will do job:
```
"SELECT * FROM Comment WHERE (comment:ancestorIds = '"
  + file.getId() + "' AND (ecm:primaryType != 'Annotation' OR annotation:xpath='file:content')"
```

Reply:
Forgot to remove this code, that won't be used in the end.